spec/features/users/user_receives_notifications_spec.rb

```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+
+describe 'notifications' do
+  describe 'user can manage notifactions' do
+    before(:each) do
+      @user = create(:user)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
+    end
+    it 'sees the user notification pane' do
+      visit dashboard_path
+      click_on("Settings")
+      expect(current_path).to eq(settings_path)
+      expect(page).to have_content("Notifications")
+      within "#notifications" do
+        expect(page).to have_content("Receive email")
+        expect(page).to have_content("Receive texts")
+        expect(page).to have_content("Rainy day notifications")
+        expect(page).to have_content("We'll let you know every morning at 8 when rain is likely in the forecast so you can consider giving your plants a moisture break.")
+        expect(page).to have_content("Frost notifications")
+        expect(page).to have_content("We'll let you know the night before when frost is likely in the forecast so you can take protective measures.")
+        expect(page).to have_content("Missed watering notifications")
+        expect(page).to have_content("We'll let you know if you forget to water your plants.")
+      end
+    end
+    it 'can update their notifications' do
+      visit settings_path
+      expect(@user.rainy_day_notifications).to eq(false)
+      expect(@user.frost_notifications).to eq(false)
+      expect(@user.receives_texts).to eq(false)
+      find(:css, "#user_rainy_day_notifications").set(true)
+      find(:css, "#user_frost_notifications").set(true)
+      click_on("Save")
+      expect(current_path).to eq(settings_path)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user.reload)
+      visit schedules_path
+      expect(@user.rainy_day_notifications).to eq(true)
+      expect(@user.frost_notifications).to eq(true)
+    end
+    it 'only lets you receive texts if you give your correct phone number' do
+      visit settings_path
+      find(:css, "#user_receives_texts").set(true)
+      click_on("Save")
+      expect(page).to have_content("Phone number can't be blank if you'd like to receive texts")
+      expect(@user.reload.receives_texts).to eq(false)
+      fill_in :user_telephone, with: "789"
+      click_on("Save")
+      expect(@user.reload.receives_texts).to eq(false)
```

Comment:
Future sad path testing could be filling in with 11 or more digits or filling in with alphabetical characters.